specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+    .SetTimeStamp(DateTime.now())
+    .AddText(L"Jill Bender", AppNotificationTextProperties().SetMaxLines(1))
+    .AddText(L"Checkout where we camped last weekend")
+    .SetAppLogoOverride(winrt::Windows::Foundation::Uri(LR"(https://unsplash.it/64?image=1027)"), ImageCrop::Circle)
+    .SetHeroImage(winrt::Windows::Foundation::Uri(LR"(https://unsplash.it/360/180?image=1043)"))
+    .AddTextBox(L"textBox", L"reply")
+    .AddButton(AppNotificationButton()
+        .AddArgument(L"action", L"reply")
+        .AddArgument(L"threadId", 92187)
+        .SetIcon(winrt::Windows::Foundation::Uri(LR"(Assets/Icons/send.png)"))
+        .SetInputId(L"textBox"))
+    .GetXml();
+```
+
+Result from GetXml():
+
+```c#
+<toast launch="action=openThread&amp;threadId=92187" displayTimestamp="2022-06-19T08:30:00Z">
+    <visual>
+        <binding template="ToastGeneric">
+            <text hint-maxLines="1">Jill Bender</text>
+            <text>Check out where we camped last weekend! It was incredible, wish you could have come on the backpacking trip!</text>
+            <image placement="appLogoOverride" hint-crop="circle" src="https://unsplash.it/64?image=1027"/>
+            <image placement="hero" src="https://unsplash.it/360/180?image=1043"/>
+        </binding>
+    </visual>
+    <actions>
+        <input id="textBox" type="text" placeHolderContent="reply"/>
+        <action
+          content="Send"
+          imageUri="Assets/Icons/send.png"
+          hint-inputId="textBox"
+          arguments="action=reply&amp;threadId=92187"/>
+    </actions>
+</toast>
+```
+
+## Limitations
+
+-   AppNotificationScenario::Urgent is only supported for to builds 19041 (20H1) and above. Using
+    this attribute on lower builds will not prevent the AppNotification from being constructed, but
+    developers will receive a debug warning that the attribute is not supported.
+
+# Text
+
+The Text component sets up the xml for a \<text\> element. Developers can define the content,
+language, and placement of the text on the AppNotification.
+
+**WinAppSDK 1.2 \<text\> Schema**:
+
+```c#
+<text lang? = string
+      placement? = "attribution"
+      hint-callScenarioCenterAlign? =
+      hint-maxLines? = integer />
+```
+
+These attributes are abstracted with the AppNotificationTextProperties component APIs:
+
+-   SetLanguage(String language)
+-   SetHintMaxLines(Int32)
+-   UseCallScenarioAlign()
+
+Below are some example usages:
+
+![Text Example](TextExample.png)
+
+AppNotificationContent Code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .SetAttributionText(L"Attribution Text")
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+      <text placement="attribution">Attribution Text</text>
+    </binding>
+  </visual>
+</toast>
+```
+
+Developers can use UseCallScenarioAlign() to center the text like an incoming call. This feature
```

Comment:
The UseCallScenarioAlign name doesn't really say what it does.
CenterText() would be more accurate (maybe in 2 years we change how incoming calls are shown, but we still want those notifications to be centered?)